extensions/docker/src/extension.ts

```diff
@@ -159,6 +159,7 @@ function initProvider(extensionContext: extensionApi.ExtensionContext) {
   });
 
   containerProviderConnection = {
+    id: 'container',
```

Comment:
Hello, I don't understand why we have id here being 'container' or 'kubernetes' as the type of object is already ContainerConnection or KubernetesConnection

Reply:
My reasoning.
To avoid duplicating the lifecycle methods i extended them to accept `KubernetesConnection ` as well. When it comes to decide which function to call i need to have a way to differentiate one object to another. 
Based on my knowledge I could check if the object contains a property to recognize it. 
`ContainerConnection ` and `KubernetesConnection` are equals except for the property type. 
So i could do 
`if 'type' in object` but it seems to weak as a check as you could add the type property to the KubernetesConnection interface tomorrow and face a misbehavior.
If we want to keep abstraction maybe we should have a generic interface and extend it? Any other way to achieve it?

Reply:
I think instanceof can be used only when instantiating a class, as we deal with an object i see this error `'ContainerProviderConnection' only refers to a type, but is being used as a value here`. 

regarding typeguard, I checked and by looking at this example i think I come back to my previous problem 
```
const isApple = (f: Fruit): f is IApple => {
    return (f as IApple).color !== undefined
}
```
i need a property to identify one type over another. Am i missing something?

Reply:
@benoitf ^